### DISCUSSION TITLE
"resolve page components" taking 500ms

### DISCUSSION THREAD

**Discussion by bitttttten** [vercel/next.js#67990] [App Router]
I am hosting a Next.JS app on Vercel and we're seeing pages that take 2.5s to render, i.e.

![Screenshot 2024-07-20 at 11 50 01](https://github.com/user-attachments/assets/f0b6cd59-1658-4cfe-b4d3-629380ab5100)

What's the best way to debug this kind of thing? We are on the Pro tier of Vercel so perhaps it's something out of our control although I just wanted to open up the discussion to see if anyone else has seem similar issues and has been able to solve them.

**Comment by aarondunphy** [comment]
I have also noticed a similar thing with our application we are hosting on Vercel. Tried debugging the Next.js internals before and it appears they have caching around the resolve page components function which is why most of the time it's instant. I assume it's made worse from the serverless function starting up fresh so there's no cache.

Like you say, we need a way to debug this so we can figure out where/why this is happening. It doesn't appear slow on basic next.js apps so I assume it's something on our end but at a loss currently.

We are enterprise customers on Vercel and we've raised a support case with them recently so will report back if we hear anything.

**Reply by ifeltsweet** [reply]
Seeing similar thing on our side. Would love to hear what Vercel team comes back with.

**Reply by zipme** [reply]
Hi, did you hear anything back from Vercel?

**Reply by aarondunphy** [reply]
They said to increase CPU function size... We tried this and it didn't help at all but haven't reached back out yet as it's too painful going in circles with their support.. will be looking to move away from the complete lack of help here.

**Reply by john-privy** [reply]
Any updates since here?

**Reply by Alexnortung** [reply]
we also have some issues with this, any updates?
